### PR TITLE
INTYGFV-14695: Added launchId to react-frontend to handle Cerner post…

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
… jump.

### CODE REVIEW

**Comment by SamuelTilly** [comment]
@ChrilleSigerhed first PR to the React codebase? Looks good! 🥳 🎊 🎉

File: packages/webcert/src/components/error/errorPageContent/InvalidLaunchIdError.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+export const TIMEOUT_TITLE = 'Intyget kan inte visas'
```
Comment (mww-sopra-steria) [inline packages/webcert/src/components/error/errorPageContent/InvalidLaunchIdError.tsx:3]: Maybe rename the constants to something else than "TIMEOUT" :-)

File: packages/webcert/src/store/api/apiMiddleware.ts [hunk 2]
```diff
@@ -28,8 +29,13 @@ const handleApiCallBegan: Middleware = ({ dispatch }: MiddlewareAPI) => () => as
       method,
       data,
       withCredentials: true,
+      headers: addLaunchIdIfPresent(),
     })
 
+    if (launchIdPresentInResponse(response) && !sessionStorage.getItem('launchId')) {
```
Comment (mww-sopra-steria) [inline packages/webcert/src/store/api/apiMiddleware.ts:35]: I believe this code should belong to the userMiddleware. The launchId is part of the user and is not generic logic. If the user changes this logic need to change.

File: packages/webcert/src/store/api/apiMiddleware.ts [hunk 3]
```diff
@@ -28,8 +29,13 @@ const handleApiCallBegan: Middleware = ({ dispatch }: MiddlewareAPI) => () => as
       method,
       data,
       withCredentials: true,
+      headers: addLaunchIdIfPresent(),
```
Comment (mww-sopra-steria) [inline packages/webcert/src/store/api/apiMiddleware.ts:32]: Maybe rename "addLaunchIdIfPresent()" to something "getHeaders()" and then within the method add launchId if it exists?

File: packages/webcert/src/store/api/apiMiddleware.ts [hunk 4]
```diff
@@ -69,7 +85,9 @@ function createApiError(api: string, response: AxiosResponse | undefined, altMes
   if (!response) {
     return { api, errorCode: 'UNKNOWN_INTERNAL_PROBLEM', message: altMessage }
   }
-
+  if (response.data && response.data.INVALID_LAUNCHID) {
```
Comment (mww-sopra-steria) [inline packages/webcert/src/store/api/apiMiddleware.ts:88]: Adjust based on the changes ask for in the Webcert PR. If you change in Webcert to use errorCode and message in the response, this code is redundant and the existing code on line 91 will work out of the box.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
A few comments to look at before merging and I'm not seeing any unit tests. Is it possible to do something here?

**Review by mww-sopra-steria — APPROVED** [review]
Nice!

**Pull request merged**